### PR TITLE
Habit was commented notification

### LINKED ISSUE #6574: Habit was commented notification
**As a** user **I want to** be notified when my habits are commented **so that** I can reply to a comment sooner

**Preconditions**
1. User 1 adds  a habit and User 2 leaves a comment for the added habit

### Acceptance Criteria

- When a new comment is left under the user's new habit, the user must get a new notification about it
- A notification counter value is increased by one
- User should see a corresponding notification record when he views the most recent notifications popup or All notifications page:
       - [Username] commented on your habbit [Habit name]. [Date and Time]
         (ex. "Maria commented on your habit Do not use disposable packages. Today 10:00 a.m")
       - [Username] - user who left a message
       - If more than one user left a comment by the moment the user sees this notification, then the usernames of 2 last 
       commentators and "other users" should be instead (ex, "Maria, Olha and other users commented on your habit Do not 
       use disposable packages. Today 10:00 a.m")
      - If the same user left more than one comment on the same habit by the moment the user sees this notification, then user 
         should see only the latest one in the notification record
      - [Habit name] - name of the event. Should be a link, should be cut if too long (?)
      - [Date and time] - Date and time when the comment was left.
         Date format is DD.MM.YYYY
        Time format is HH:MM XM
        If the day = Today, then use "Today" word instead of calendar date
        If the day = Today - 1, then use "Yesterday" word instead of calendar date

**Epic link**
E.g.: Epic #6332

**Labels to be added**
"User story"

### PR DESCRIPTION
**Pull request by K4sik**
# GreenCity PR
Feature: Commenting on Habits with Tagging, Replies, and Notifications

* Users can leave comments on specific habits.
* Users can tag others in the comments using "@" mentions.
* Users can reply directly to specific comments in a threaded format.
* Notifications are sent to habit owners and tagged users when a new comment or reply is added.

## Issue Link :clipboard:
#6574 

## Added
* Added new methods in CommentServiceImpl: createNotification(), createCommentNotification(). createCommentReplyNotification(), sendNotificationToTaggedUser(), getArticleType() and getNotificationType();
* Added new enum CommentActionType;
* Added test cases for every method.

## Changed
* Updated HabitCommentController by adding Localization;
* Updated ErrorMessager file by adding UNSUPPORTED_ARTICLE_TYPE and UNSUPPORTED_ACTION_TYPE fields to it;
* Updated notification.properties and notification_uk.properties files by adding HABIT_COMMENT_TITLE, HABIT_COMMENT, HABIT_COMMENT_REPLY_TITLE, HABIT_COMMENT_REPLY, HABIT_COMMENT_USER_TAG_TITLE, and HABIT_COMMENT_USER_TAG fields to them;
* Updated NotificationType file by adding HABIT_COMMENT, HABIT_COMMENT_REPLY, and HABIT_COMMENT_USER_TAG enum elements.

# PR Checklist Forms

- [ ] Code is up-to-date with the dev branch.
- [X] You've successfully built and run the tests locally.
- [X] New or updated unit tests validate the change.
- [X] JIRA/ Github Issue number & title in PR title (ISSUE-#6574)
- [X] This template is filled (above this section).
- [X] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells or duplication
- [X] NEED_REVIEW and READY_FOR_REVIEW labels are added.
- [X] All files reviewed before sending to reviewers

### CODE REVIEW

**Comment by holotsvan** [comment]
image

1 issue

**Review by KizerovDmitriy — APPROVED** [review]

File: service/src/main/java/greencity/service/CommentServiceImpl.java [hunk 1]
```diff
@@ -3,36 +3,26 @@
 import greencity.achievement.AchievementCalculation;
 import greencity.constant.ErrorMessage;
 import greencity.dto.PageableDto;
-import greencity.dto.comment.AddCommentDtoRequest;
-import greencity.dto.comment.AddCommentDtoResponse;
-import greencity.dto.comment.CommentAuthorDto;
-import greencity.dto.comment.CommentDto;
+import greencity.dto.comment.*;
```
Comment (lukovskyj) [inline service/src/main/java/greencity/service/CommentServiceImpl.java]: do not use wildcard imports

Comment (holotsvan) [inline service/src/main/java/greencity/service/CommentServiceImpl.java:6]: don't use wildcard imports

**Review by holotsvan — APPROVED** [review]

File: service/src/main/java/greencity/service/CommentServiceImpl.java (hunk 1 above)
Comment (holotsvan) [reply service/src/main/java/greencity/service/CommentServiceImpl.java]: @K4sik 

In intellij you could go to settings -> editor -> code style -> java

make sure Use single class import is enabled
in class count to use import with '\*' and names count to use static import with '\*' set some big number (e.g 999) and intellij idea would import each class separately.

**Review by lukovskyj — APPROVED** [review]

**Review by softservedata — APPROVED** [review]

**Review by IliaWithHat — APPROVED** [review]

**Pull request merged**